esi_leap/manager/service.py

```diff
@@ -130,6 +140,62 @@ def _expire_offers(self):
                     offer.status = statuses.ERROR
                     offer.save()
 
+    def _notify_lease_expire(self):
+        LOG.info('Checking for leases expire soon')
+        leases = lease_obj.Lease.get_all(
+            {'status': [statuses.ACTIVE, statuses.CREATED,
+                        statuses.WAIT_EXPIRE, statuses.WAIT_FULFILL]},
+            self._context)
+        now = timeutils.utcnow()
+        for lease in leases:
+            if lease.end_time - now <= datetime.timedelta(1):
+                try:
+                    LOG.info('Lease %s is expiring in 1 day', lease.uuid)
+                    self._send_email(lease)
+                except Exception as e:
+                    LOG.info('Error sending lease expire '
+                             'email notification: %s: %s' %
+                             (type(e).__name__, e))
+
+    def _send_email(self, lease):
```

Comment:
I'd put this into its own utility function which takes in the 'to_email' and the body of the email

Reply:
done